### ISSUE TITLE
`from_string` for `Dervive` in `no-std`

### ISSUE THREAD

**Issue by michalkucharczyk** [paritytech/substrate#14081]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Motivation

It is a follow-up of: https://github.com/paritytech/substrate/pull/13027

### Request

Proper implementation of `from_string` functions for `Derive` is still missing:
https://github.com/paritytech/substrate/pull/13027#discussion_r1185370246, 
https://github.com/paritytech/substrate/commit/0e00dd62ea86813adb6eeff90581575abcb24f47

### Are you willing to help with this request?

Yes!

**Comment by koute** [comment]
Replying to @michalkucharczyk your comment from the PR:

> Not sure what to do here: we could try to parse w/o regex

The regexps are not that complex; I'd probably just parse these by hand. Should be easy enough to make sure it's correct by just fuzzying the new implementation against the regex.

**Comment by bkchr** [comment]
> The regexps are not that complex; I'd probably just parse these by hand. Should be easy enough to make sure it's correct by just fuzzying the new implementation against the regex.

Yeah, I also already proposed the same to @michalkucharczyk. Some easy string search should be enough.
